### DISCUSSION TITLE
Vercel Analytics API VERCEL_ANALYTICS_ID

### DISCUSSION THREAD

**Discussion by jamezening** [vercel/vercel#8653] [Help] [answered]
Hi,

I have an Astro project deployed to Vercel.
(super performant deployment combination by the way 🚀)

I wish to hook in the [Vercel Analytics API](https://vercel.com/docs/concepts/analytics/api#) in the project and am trying to trackdown the `VERCEL_ANALYTICS_ID`.

from the link:
> This value is exposed during the build and can be accessed by process.env.VERCEL_ANALYTICS_ID inside Node.js.
I have tried logging `VERCEL_ANALYTICS_ID` and checking in the deployments log, but can't track the value down.

Can anyone share any tips / their experiences?

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @vanilla-james! You won't be able to use `vercel pull` to bring the analytics ID to your local environment variables. It is included during `vercel build` or when the project is deployed on Vercel. You do not need to manually add this as an environment variable.

There is a small note about it in the [Vercel Analytics API documentation](https://vercel.com/docs/concepts/analytics/api#getting-started): 
>`vercel pull` does not pull `VERCEL_ANALYTICS_ID` as the Vercel Analytics ID environment variable is inlined during the build process. It is not part of your project Environment Variables, which can be pulled locally using Vercel CLI.

**Reply by jamezening** [reply]
@amyegan 
Thank you for your many answers and contribution to the Vercel discussions 😊

Before I understood how this worked, I had tried logging the value during the deployment, and it showed as `undefined` which added to my own confusion.

<img width="865" alt="image" src="https://user-images.githubusercontent.com/6829451/193183072-5d4e394e-80ff-43c8-84a6-9874825a5dd2.png">

Based on your answer above, I will just confidently use the `VERCEL_ANALYTICS_ID`, knowing that it will be available during deployment.

I am also having a bit of trouble hooking up Analytics in Astro, but that is a question for another thread.
Thank you again.
